field/packer_unpacker.go

```diff
@@ -51,3 +51,67 @@ func (u defaultUnpacker) Unpack(packedFieldValue []byte, spec *Spec) ([]byte, in
 
 	return value, read + prefBytes, nil
 }
+
+type Track2Packer struct{}
+
+// This is a custom packer for Track2 Data. Some specifications require the length
+// to be equal to the length of the pre-padded value.
+func (p Track2Packer) Pack(value []byte, spec *Spec) ([]byte, error) {
+	data := value
+
+	// Only pad if the length is odd. If so, just add
+	// one pad character, so tell the Pad function that
+	// the length we want is +1 to what the value is
+	if spec.Pad != nil && len(value)%2 != 0 {
+		data = spec.Pad.Pad(data, len(value)+1)
+	}
+
+	packed, err := spec.Enc.Encode(data)
+	if err != nil {
+		return nil, fmt.Errorf("failed to encode content: %w", err)
+	}
+
+	// Encode the length to that of the original string, not the potentially
+	// padded length
+	packedLength, err := spec.Pref.EncodeLength(spec.Length, len(value))
+	if err != nil {
+		return nil, fmt.Errorf("failed to encode length: %w", err)
+	}
+
+	return append(packedLength, packed...), nil
+}
+
+type Track2Unpacker struct{}
```

Comment:
You know, I'm not sure it will make any difference. The question is: how would someone look for such a packer/unpacker?

Reply:
Would it make more sense to have a `packer` package?

Reply:
I would try to avoid it for as long as possible. The packer for track2 is the first candidate for such, but let's wait for some more. For now we can keep it here or move into track2.go.